lua/cli_flash.lua

```diff
@@ -283,6 +283,22 @@ addJtagResetArg(tParserCommandErase)
 addJtagKhzArg(tParserCommandErase)
 addSecureArgs(tParserCommandErase)
 
+-- smart erase
+local tParserCommandSmartErase = tParser:command('smart_erase se', 'Erase area inside SPI-flash with optimised erase block sizes'):target('fCommandSmartEraseSelected')
+-- required_args = {"b", "u", "cs", "s", "l"}
+addBusOptionArg(tParserCommandSmartErase)
+addUnitOptionArg(tParserCommandSmartErase)
+addChipSelectOptionArg(tParserCommandSmartErase)
+addStartOffsetArg(tParserCommandSmartErase)
+addLengthArg(tParserCommandSmartErase)
+-- optional_args = {"p", "t", "jf", "jr"}
+addPluginNameArg(tParserCommandSmartErase)
+addPluginTypeArg(tParserCommandSmartErase)
+addJtagResetArg(tParserCommandSmartErase)
+addJtagKhzArg(tParserCommandSmartErase)
+addSecureArgs(tParserCommandSmartErase)

```

Comment:
There's nothing wrong with this, but do we want an extra erase command, or rather an option to the normal erase command? What about the erase operation that may be done before flashing data?

Reply:
Ok for now, may be turned into an option later on or even replace the standard erase.